### PR TITLE
Send an error code on polymer command run error

### PR DESCRIPTION
**Pull request by pmaudsley**
PR request to solve issue https://github.com/Polymer/polymer-cli/issues/315

Adding a process.exit(1) on a error from cli.run() to prevent error code 0 returned for failed polymer test runs

### CODE REVIEW

File: bin/polymer.js [hunk 1]
```diff
@@ -26,5 +26,6 @@ resolve('polymer-cli', {basedir: process.cwd()}, function(error, path) {
   let lib = path ? require(path) : require('..');
   let args = process.argv.slice(2);
   let cli = new lib.PolymerCli(args);
-  cli.run();
+  cli.run()
+  .then(null, () => process.exit(1));
```
Comment (mattyclarkson) [inline bin/polymer.js]: You can use `catch` rather than the second parameter of `then`:

```
cli.run().catch(() => process.exit(1));
```

Comment (FredKSchott) [inline bin/polymer.js]: +1, using `then()` like this is fine, but we use `catch()` throughout the rest of the project. Lets stick to that here.

Also, can you please log a message before exit? Check out the `process.on('unhandledRejection', ...` handler in `polymer-cli.ts` for an example of what you'll need.

**Comment by FredKSchott** [comment]
@pmaudsley thanks for the PR! I can't believe we missed this, thanks for adding.

**Comment by abdonrd** [comment]
Thanks for the PR!

**Comment by pmaudsley** [comment]
I've made the changes as suggested. 

Is it OK that this logger repeats the previous logs i.e.:

```
1 failed tests

error:   1 failed tests
```

**Comment by FredKSchott** [comment]
@pmaudsley where are you seeing that "1 failed tests" error message? Do you have steps to reproduce so I can understand better?

File: bin/polymer.js [hunk 2]
```diff
@@ -26,5 +29,9 @@ resolve('polymer-cli', {basedir: process.cwd()}, function(error, path) {
   let lib = path ? require(path) : require('..');
   let args = process.argv.slice(2);
   let cli = new lib.PolymerCli(args);
-  cli.run();
+  cli.run().catch(err => {
+    logger.error(err);
```
Comment (FredKSchott) [inline bin/polymer.js]: Can you include a message here? It's helpful in debugging to know where the error was caught & logged. See polymer-cli.ts `uncaughtRejection` handler for example.

**Comment by FredKSchott** [comment]
Superseded by #321

**Pull request closed without merging**